From 07a2e0ce1800768320985136d1a73ee95cf3a5b1 Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Sat, 7 Mar 2020 14:49:09 +0100
Subject: [PATCH 2/5] Failing to create facedata shouldn't be fatal

Some Pie vendors create it on their own, so SELinux would deny that
Also not all devices have face unlock anyway

See https://github.com/phhusson/treble_experimentations/issues/1119
---
 vold_prepare_subdirs.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/vold_prepare_subdirs.cpp b/vold_prepare_subdirs.cpp
index 60e82f50a..c1ded02e7 100644
--- a/vold_prepare_subdirs.cpp
+++ b/vold_prepare_subdirs.cpp
@@ -205,7 +205,7 @@ static bool prepare_subdirs(const std::string& volume_uuid, int user_id, int fla
             }
             auto facedata_path = vendor_de_path + "/facedata";
             if (!prepare_dir(sehandle, 0700, AID_SYSTEM, AID_SYSTEM, facedata_path)) {
-                return false;
+                LOG(ERROR) << "Failed preparing folder for de facedata";
             }
         }
     }
@@ -254,7 +254,7 @@ static bool prepare_subdirs(const std::string& volume_uuid, int user_id, int fla
             auto vendor_ce_path = android::vold::BuildDataVendorCePath(user_id);
             auto facedata_path = vendor_ce_path + "/facedata";
             if (!prepare_dir(sehandle, 0700, AID_SYSTEM, AID_SYSTEM, facedata_path)) {
-                return false;
+                LOG(ERROR) << "Failed preparing folder for de facedata";
             }
         }
     }
-- 
2.34.1

